### PR TITLE
Added ability to add QuiverKey object to Legend

### PR DESCRIPTION
**Pull request by Sokojoe**
Fixes https://github.com/matplotlib/matplotlib/issues/16664

## PR Summary

A QuiverKey object can now be added to the legend as a handle.  

Co-authored by @wesleyma2, @rayafarhadi and @gesarki

## PR Checklist

- [ x ] Has Pytest style unit tests
- [ x ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ x ] New features are documented, with examples if plot related
- [ x ] Documentation is sphinx and numpydoc compliant
- [ x ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ x ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by Sokojoe** [comment]
It seems some of the ci is failing. We aren't quite sure where exactly in the docs the failure is occuring, or how we can test this. Could somebody help us out?

**Comment by tacaswell** [comment]
I power-cycled to try and get a new merge commit against master, but then belatedly realized that circle runs on your commit, not the synthetic merge commit :facepalm: .

There were some changes with sphinx3 that required slight changes to our docs build (which is likely the source of your problems).  The next thing to try is to rebase on current master and then `git push --force-with-lease` to your branch.

File: lib/matplotlib/legend_handler.py [hunk 1]
```diff
@@ -728,3 +729,35 @@ def create_artists(self, legend, orig_handle,
...
+
+        # we force labelpos to be 'N' so that quiverkey draws the center
+        # of the arrow at (q.X, q.Y) and q.angle rotates the arrow
+        # around the midpoint
+        q.labelpos = 'N'
+        q._init()
+
+        # Hide original vector + label
+        try:
+            orig_handle.remove()
```
Comment (tacaswell) [inline lib/matplotlib/legend_handler.py:760]: This is a very surprising thing to do in this part of the code.

**Comment by tacaswell** [comment]
Thank you for working on this!  However I do not think this is the right apporach.

See comment at https://github.com/matplotlib/matplotlib/issues/16664#issuecomment-613117125 .  Implementing a custom handler is the right path, but it should create a `QuiverKey` object to use in the legend for a `Quiver`, not transport a `QuiverKey` from the Axes -> the legend.

Put another way, the goal is to use a `QuiverKey` in the legend, not to create a legend entry for a QuiverKey.

**Comment by rayafarhadi** [comment]
I'm not sure I'm understanding your change request fully. Can you clarify the difference between "use a QuiverKey in the legend" and "create a legend entry for a QuiverKey"?

**Comment by timhoffm** [comment]
> I'm not sure I'm understanding your change request fully. Can you clarify the difference between "use a QuiverKey in the legend" and "create a legend entry for a QuiverKey"?

You have implemented
```
qk = ax.quiverkey(Q, 0.9, 0.8, ...)
ax.legend([qk])
```

But we want
```
q = ax.quiver(...)
ax.legend([q])
```

and internally the legend handler mechanism must be implemented to create a `QuiverKey` as the proxy artist used in the legend for the given `Quiver` object `q`.

**Comment by QuLogic** [comment]
AFAICT, this was the wrong approach, and as it appears to be orphaned, there's no point to keeping the PR open.

**Pull request closed without merging**